pinot-controller/src/main/java/com/linkedin/pinot/controller/validation/StorageQuotaChecker.java

```diff
@@ -118,21 +118,23 @@ public QuotaCheckerResponse isSegmentStorageWithinQuota(@Nonnull File segmentFil
     _controllerMetrics.setValueOfTableGauge(tableName, ControllerGauge.OFFLINE_TABLE_ESTIMATED_SIZE,
         tableSubtypeSize.estimatedSizeInBytes);
 
-    // incomingSegmentSizeBytes is compressed data size for just 1 replica. Dictionary and indices sizes aren't included.
+    // incomingSegmentSizeBytes is compressed data size for just 1 replica.
     long estimatedFinalSizeBytes =
         tableSubtypeSize.estimatedSizeInBytes - existingSegmentSizeBytes + (incomingSegmentSizeBytes * numReplicas);
     if (estimatedFinalSizeBytes <= allowedStorageBytes) {
       String message = String.format(
-          "Estimated size: %d bytes is within the total configured quota of %d bytes for table %s.\nStorage quota: %s (%d bytes). Number of replicas: %d.\nExisting total usage among all replicas: %d bytes.\nIncoming compressed segment size: %d bytes.",
-          estimatedFinalSizeBytes, allowedStorageBytes, tableName, quotaConfig.getStorage(),
-          quotaConfig.storageSizeBytes(), numReplicas, tableSubtypeSize.estimatedSizeInBytes, incomingSegmentSizeBytes);
+          "Estimated size: %d bytes ( = current usage among all replicas: %d bytes - (existing segment sizes among all replicas: %d bytes) + (incoming compressed segment size: %d bytes * number of replicas: %d)) is within the total configured quota of %d bytes ( = configured quota: %d bytes * number of replicas: %d ) for table %s.",
+          estimatedFinalSizeBytes, tableSubtypeSize.estimatedSizeInBytes, existingSegmentSizeBytes,
+          incomingSegmentSizeBytes, numReplicas, allowedStorageBytes, quotaConfig.storageSizeBytes(), numReplicas,
+          tableName);
       LOGGER.info(message);
       return new QuotaCheckerResponse(true, message);
     } else {
       String message = String.format(
-          "Estimated size: %d bytes exceeds the total configured quota of %d bytes for table %s.\nStorage quota: %s (%d bytes). Number of replicas: %d.\nExisting total usage among all replicas: %d bytes.\nIncoming compressed segment size: %d bytes.",
-          estimatedFinalSizeBytes, allowedStorageBytes, tableName, quotaConfig.getStorage(),
-          quotaConfig.storageSizeBytes(), numReplicas, tableSubtypeSize.estimatedSizeInBytes, incomingSegmentSizeBytes);
+          "Estimated size: %d bytes ( = current usage among all replicas: %d bytes - (existing segment sizes among all replicas: %d bytes) + (incoming compressed segment size: %d bytes * number of replicas: %d)) exceeds the total configured quota of %d bytes ( = configured quota: %d bytes * number of replicas: %d ) for table %s.",
```

Comment:
Lets first tell the user about the problem and then elaborate. "Storage quota exceeded. Estimated size....".

On thinking more, some of the error cases happen where the table has already exceeded size (indexes enabled prior to push etc). In that case, I think we can probably say, "Table already over quota. Estimated uncompressed size of all replicas > %d (configured value X * num replicas). Check if indexes were enabled recently and adjust table quota accordingly".